File: Python/bijele.py
'''
 *
 *	Problem:		Bijele
 *	Problem ID:	bijele
 *	url:				https://open.kattis.com/problems/bijele
 *	Solution:		Python
 *
'''

import itertools

correctAmounts = [1, 1, 2, 2, 2, 8]
pieceAmounts = [ int(pieceAmount) for pieceAmount in input().split(" ")]

difference_object = zip(correctAmounts, pieceAmounts)

difference = []
for i, j, in difference_object:
  difference.append(str(i - j))

print(" ".join(difference))

File: Ruby/bijele.rb
=begin
 *
 *	Problem:		Bijele
 *	Problem ID:	bijele
 *	url:				https://open.kattis.com/problems/bijele
 *	Solution:		Ruby
 *
=end

$correct_amounts = [1, 1, 2, 2, 2, 8]

class Solver

	def solve
		line = gets.chomp
		piece_amounts= line.split(" ").map(&:to_i)

		result = ""
		$correct_amounts.each_with_index do |amount, index|
			difference = amount - piece_amounts[index]
			result += difference.to_s + " "
		end		
		
		puts result
	end
	
end

solver = Solver.new
solver.solve